core/src/main/java/io/snyk/plugins/artifactory/scanner/PackageValidator.java

```diff
@@ -2,108 +2,57 @@
+
+    int countAboveThreshold = artifact.getVulnSummary().getCountAtOrAbove(severityThreshold);
+    if (countAboveThreshold > 0) {
+      LOG.debug("Found {} vulnerabilities in {} returning 403", countAboveThreshold, artifact);
+      throw new CancelException(format("Artifact has vulnerabilities with at least %s severity. %s", severityThreshold.getSeverityLevel(), artifact), 403);
     }
   }
 
-  private void validateLicenseIssues(TestResult testResult, RepoPath repoPath) {
-    final String licensesForceDownloadProperty = ISSUE_LICENSES_FORCE_DOWNLOAD.propertyKey();
-    final String licensesForceDownload = repositories.getProperty(repoPath, licensesForceDownloadProperty);
-    final boolean forceDownload = "true".equalsIgnoreCase(licensesForceDownload);
-    if (forceDownload) {
-      LOG.debug("Allowing download. Artifact Property \"{}\" is \"true\". {}", repoPath, licensesForceDownloadProperty);
+  private void validateLicenseIssues(MonitoredArtifact artifact) {
+    if (artifact.getIgnores().shouldIgnoreLicenseIssues()) {
+      LOG.debug("Allowing download because license issues are ignored");
       return;
     }
 
-    Severity licensesThreshold = Severity.of(configurationModule.getPropertyOrDefault(PluginConfiguration.SCANNER_LICENSE_THRESHOLD));
-    if (licensesThreshold == Severity.LOW) {
-      if (!testResult.issues.licenses.isEmpty()) {
-        LOG.debug("Found license issues in {} returning 403", repoPath);
-        throw new CancelException(format("Artifact has license issues. %s", repoPath), 403);
-      }
-    } else if (licensesThreshold == Severity.MEDIUM) {
-      long count = testResult.issues.licenses.stream()
-        .filter(vulnerability -> vulnerability.severity == Severity.MEDIUM || vulnerability.severity == Severity.HIGH)
-        .count();
-      if (count > 0) {
-        LOG.debug("Found {} license issues in {} returning 403", count, repoPath);
-        throw new CancelException(format("Artifact has license issues with medium or high severity. %s", repoPath), 403);
-      }
-    } else if (licensesThreshold == Severity.HIGH) {
-      long count = testResult.issues.licenses.stream()
-        .filter(vulnerability -> vulnerability.severity == Severity.HIGH)
-        .count();
-      if (count > 0) {
-        LOG.debug("Found {} license issues in {} returning 403", count, repoPath);
-        throw new CancelException(format("Artifact has license issues with high severity. %s", repoPath), 403);
-      }
+    Severity severityThreshold = settings.getLicenseSeverityThreshold();
+
+    int countAboveThreshold = artifact.getLicenseSummary().getCountAtOrAbove(severityThreshold);
+    if (countAboveThreshold > 0) {
+      LOG.debug("Found {} license issues in {} returning 403", countAboveThreshold, artifact);
+      throw new CancelException(format("Artifact has license issues with at least %s severity. %s", severityThreshold.getSeverityLevel(), artifact), 403);
```

Comment:
This new exception is missing the `repoPath`, will this change affect our/clients ability to figure out from the logs in which repo the problems occurred?

Reply:
Ahhh great spot, I was going to have the path exposed via `artifact.toString()` but I forgot - I'll fix that before I merge.